src/main/java/org/sunbird/extension/user/impl/UserProviderRegistryImpl.java

```diff
@@ -0,0 +1,297 @@
+    userIdMap.put(SunbirdExtensionConstants.USER_TYPE, "teacher");
+    userIdMap.put(SunbirdExtensionConstants.X_AUTHENTICATED_USER_TOKEN, "accessToken");
+
+    return readUser(userIdMap);
+  }
+
+  @Override
+  public void update(Map<String, Object> userProfileMap) {
+
+    // hardcoded to teacher till userType enhancement is done
+    userProfileMap.put(SunbirdExtensionConstants.USER_TYPE, "teacher");
+
+    updateUser(userProfileMap);
+  }
+
+  @Override
+  public void delete(Map<String, Object> userProfileMap) {
+    // to be implemented
+  }
+
+  private Map<String, String> getHeader(String accessToken) {
+    Map<String, String> headers = new HashMap<>();
+    headers.put(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE);
+    headers.put(SunbirdExtensionConstants.X_AUTHENTICATED_USER_TOKEN, accessToken);
+    return headers;
+  }
+
+  private void addUser(Map<String, Object> userProfileMap) {
+
+    String accessToken = getAccessToken(userProfileMap);
+    Map<String, Object> userMap = getUserMapForWrite(userProfileMap);
+
+    ResponseData<String> responseData = null;
+    try {
+      responseData =
+          client.addEntity(
+              new RequestData<>(mapper.writeValueAsString(userMap)), getHeader(accessToken));
+    } catch (JsonProcessingException | TransformationException e) {
+      ProjectLogger.log(
+          "UserProviderRegistryImpl:addUser : User Registry Add Entity failed ==> "
+              + e.getStackTrace(),
+          LoggerEnum.ERROR.name());
+      throwUserRegistryAddEntityException();
+    }
+
+    Map<String, Object> responseMap = getResponseMap(responseData);
+    Map<String, Object> paramsMap =
+        (Map<String, Object>) responseMap.get(SunbirdExtensionConstants.PARAMS);
+
+    if (SunbirdExtensionConstants.STATUS_SUCCESS.equalsIgnoreCase(
+        (String) paramsMap.get(SunbirdExtensionConstants.STATUS))) {
+      Map<String, Object> resultMap =
+          (Map<String, Object>) responseMap.get(SunbirdExtensionConstants.RESULT);
+      userProfileMap.put(
+          SunbirdExtensionConstants.REGISTRY_ID, resultMap.get(SunbirdExtensionConstants.ENTITY));
+    } else {
+      String errMsg = (String) paramsMap.get(SunbirdExtensionConstants.ERR_MSG);
+      ProjectLogger.log(
+          "UserProviderRegistryImpl:addUser : User Registry Add Entity failed - " + errMsg,
+          LoggerEnum.ERROR.name());
+      throwUserRegistryAddEntityException();
+    }
+  }
+
+  private Map<String, Object> readUser(Map<String, Object> userIdMap) {
+
+    String accessToken = getAccessToken(userIdMap);
+    ResponseData<String> responseData = null;
+    String registryId = getRegistryId(userIdMap);
+
+    try {
+      responseData = client.readEntity(new URI(registryId), getHeader(accessToken));
+    } catch (TransformationException | URISyntaxException e) {
+      ProjectLogger.log(
+          "UserProviderRegistryImpl:readUser : User Registry Read Entity failed ==> "
```

Comment:
"UserProviderRegistryImpl:readUser: User registry client read entity resulted in exception = "